lib/ansible/modules/cloud/amazon/ec2_lc.py

```diff
@@ -45,95 +52,100 @@
+        device type) must be specified for a new volume or a root volume, and may be passed for a snapshot volume. For any volume, a volume size less
+        than 1 will be interpreted as a request not to create the volume.
   user_data:
     description:
-      - opaque blob of data which is made available to the ec2 instance. Mutually exclusive with I(user_data_path).
-    required: false
+      - Opaque blob of data which is made available to the ec2 instance. Mutually exclusive with I(user_data_path).
   user_data_path:
     description:
       - Path to the file that contains userdata for the ec2 instances. Mutually exclusive with I(user_data).
-    required: false
     version_added: "2.3"
   kernel_id:
     description:
       - Kernel id for the EC2 instance
-    required: false
   spot_price:
     description:
       - The spot price you are bidding. Only applies for an autoscaling group with spot instances.
-    required: false
   instance_monitoring:
     description:
-      - whether instances in group are launched with detailed monitoring.
+      - Whether instances in group are launched with detailed monitoring.
     default: false
   assign_public_ip:
     description:
-      - Used for Auto Scaling groups that launch instances into an Amazon Virtual Private Cloud. Specifies whether to assign a public IP
-        address to each instance launched in a Amazon VPC.
-    required: false
+      - Used for Auto Scaling groups that launch instances into an Amazon Virtual Private Cloud. Specifies whether to assign a public IP address
+        to each instance launched in a Amazon VPC.
     version_added: "1.8"
   ramdisk_id:
     description:
       - A RAM disk id for the instances.
-    required: false
     version_added: "1.8"
   instance_profile_name:
     description:
       - The name or the Amazon Resource Name (ARN) of the instance profile associated with the IAM role for the instances.
-    required: false
     version_added: "1.8"
   ebs_optimized:
     description:
       - Specifies whether the instance is optimized for EBS I/O (true) or not (false).
-    required: false
     default: false
     version_added: "1.8"
   classic_link_vpc_id:
     description:
       - Id of ClassicLink enabled VPC
-    required: false
     version_added: "2.0"
   classic_link_vpc_security_groups:
     description:
-      - A list of security group id's with which to associate the ClassicLink VPC instances.
-    required: false
+      - A list of security group IDs with which to associate the ClassicLink VPC instances.
     version_added: "2.0"
   vpc_id:
     description:
       - VPC ID, used when resolving security group names to IDs.
-    required: false
     version_added: "2.4"
+  instance_id:
+    description:
+      - The Id of a running instance to use as a basis for a launch configuration. Can be used in place of image_id and instance_type.
+    version_added: "2.4"
+  advanced_instance_monitoring:
```

Comment:
Have made `instance_monitoring` synonymous with `advanced_instance_monitoring` so either can be set to true and it will be enabled. Could not remove `instance_monitoring` as it is legacy so thought this would be the next best alternative. Let me know if this is not what you meant.